examples/Microphysics/ex_1_saturation_adjustment.jl

```diff
@@ -0,0 +1,335 @@
+                                            eulerflux!,
+                                            wavespeed,
+                                            preflux
+                                           )
+  numbcflux!(x...) = NumericalFluxes.rusanov_boundary_flux!(x...,
+                                                            eulerflux!,
+                                                            bcstate!,
+                                                            wavespeed,
+                                                            preflux
+                                                           )
+
+
+
+  # spacedisc = data needed for evaluating the right-hand side function
+  spacedisc = DGBalanceLaw(grid = grid,
+                           length_state_vector = _nstate,
+                           flux! = eulerflux!,
+                           numerical_flux! = numflux!,
+                           numerical_boundary_flux! = numbcflux!,
+                           auxiliary_state_length = _nauxcstate,
+                           auxiliary_state_initialization! =
+                             constant_auxiliary_init!)
+
+  # This is a actual state/function that lives on the grid
+  initialcondition(Q, x...) = single_eddy!(Q, DFloat(0), x...)
+  Q = MPIStateArray(spacedisc, initialcondition)
+
+  npoststates = 9
+  v_q_liq, v_q_vap, v_q_tot, v_p, v_T, v_e_tot, v_e_int, v_e_kin, v_e_pot =
+    1:npoststates
+  postnames = ("q_liq", "q_vap","q_tot", "p", "T",
+               "e_tot", "e_int", "e_kin", "e_pot")
+  postprocessarray = MPIStateArray(spacedisc; nstate=npoststates)
+
+  writevtk("initial_condition", Q, spacedisc, statenames)
+
+  lsrk = LSRK54CarpenterKennedy(spacedisc, Q; dt = dt, t0 = 0)
+
+  io = MPI.Comm_rank(mpicomm) == 0 ? stdout : devnull
+  eng0 = norm(Q)
+  @printf(io, "----\n")
+  @printf(io, "||Q||₂ (initial) =  %.16e\n", eng0)
+
+  # Set up the information callback
+  timer = [time_ns()]
+  cbinfo = GenericCallbacks.EveryXWallTimeSeconds(10, mpicomm) do (s=false)
+    if s
+      timer[1] = time_ns()
+    else
+      run_time = (time_ns() - timer[1]) * 1e-9
+      (min, sec) = fldmod(run_time, 60)
+      (hrs, min) = fldmod(min, 60)
+      @printf(io, "----\n")
+      @printf(io, "simtime =  %.16e\n", ODESolvers.gettime(lsrk))
+      @printf(io, "runtime =  %03d:%02d:%05.2f (hour:min:sec)\n", hrs, min, sec)
+      @printf(io, "||Q||₂  =  %.16e\n", norm(Q))
+    end
+    nothing
+  end
+
+  step = [0]
+  mkpath("vtk")
+
+  cbvtk = GenericCallbacks.EveryXSimulationSteps(60) do (init=false)
+    DGBalanceLawDiscretizations.dof_iteration!(postprocessarray, spacedisc,
+                                               Q) do R, Q, QV, aux
+      @inbounds begin
+        u, w, ρ, q_tot, e_tot = preflux(Q)
+        z = aux[_c_z]
+        p = aux[_c_p]
+
+        e_int = e_tot - 1//2 * (u^2 + w^2) - grav * z
+        ts = PhaseEquil(e_int, q_tot, ρ)  # saturation adjustment happens here
```

Comment:
Why do we need it for explicit time stepping, if q_liq is part of the state?

Reply:
In the Kessler example I change it to q_eq = PhasePartition_equil(T, ρ, q_tot). In the example 1 for saturation adjustment I keep it